File: src/go/problems/TwoSum_test.go
// https://leetcode.com/problems/two-sum/
package problems

import "testing"

func TestTwoSum(t *testing.T) {
    result := TwoSum([]int{2,7,11,15}, 9)
    t.Log(result)
}

func TwoSum(nums []int, target int) []int {
    set := make(map[int]int, len(nums))

    for i, n := range nums {
        if v, ok := set[target - n]; ok {
            return []int{v, i}
        }

        set[n] = i
    }

    return []int{}
}

File: src/python/problems/two_sum_test.py
#https://leetcode.com/problems/two-sum/

from core.problem_base import *

class TwoSum(ProblemBase):
    def Solution(self, nums: list[int], target: int) -> list[int]:
        map = {}

        for i in range(len(nums)):
            if target - nums[i] in map:
                return [map[target - nums[i]], i]
            map[nums[i]] = i
            
        return []

        
if __name__ == '__main__':
    TestGen(TwoSum) \
        .Add(lambda tc: tc.Param([2,7,11,15]).Param(9).Result([0,1])) \
        .Add(lambda tc: tc.Param([3,2,4]).Param(6).Result([1,2])) \
        .Add(lambda tc: tc.Param([3,3]).Param(6).Result([0,1])) \
        .Run()
